lti_consumer/lti_1p3/consumer.py

```diff
@@ -540,19 +556,45 @@ def enable_deep_linking(
         """
         self.dl = LtiDeepLinking(deep_linking_launch_url, deep_linking_return_url)
 
-    def generate_launch_request(
-            self,
-            preflight_response,
-            resource_link
+    def enable_proctoring(
+        self,
+        attempt_number,
+        session_data=None,
+        start_assessment_url=None,
+        resource_link=None,
     ):
         """
-        Build LTI message for Deep linking launches.
+        Enable LTI 1.3 Proctoring Service.
 
-        Overrides method from LtiConsumer1p3 to allow handling LTI Deep linking messages
+        This will include the LTI Proctoring Service claims in the LTI launch message
+        and set up the required class.
         """
-        # Check if Deep Linking is enabled and that this is a Deep Link Launch
-        if self.dl and preflight_response.get("lti_message_hint") == "deep_linking_launch":
-            # Validate preflight response
+        self.proctoring = LtiProctoring(attempt_number, session_data, start_assessment_url, resource_link)
+
+    def set_proctoring_user_data(self, **kwargs):
+        """
+        Set the optional user data claims, per the Proctoring Services specification.
+        """
+        # As defined in the LTI Core specification [LTI-13], the https://purl.imsglobal.org/spec/lti/claim/roles claim
+        # is required but MAY be empty.
+        # The Start Proctoring message always initiates the proctoring process (and corresponding assessment launch)
+        # regardless of any value specified for this claim, the Proctoring Tool MUST ignore this value.
+        # If the Assessment Platform provides a non-empty value it SHOULD include
+        # http://purl.imsglobal.org/vocab/lis/v2/membership#Learner.
+        # TODO: This is hacky. The roles list should either be empty or should include #Learner. #Leaner is a
```

Comment:
:+1: for this. I think we can just add it here: https://github.com/openedx/xblock-lti-consumer/blob/master/lti_consumer/lti_1p3/constants.py#L29